### PR TITLE
feat: Unread activities information Mark as read from the left menu -MEED-701- Meeds-io/MIPs#17

### PR DESCRIPTION
**Pull request by SaraBoutej**
In this PR we add the possibility to mark all unread items of space application as read from the left navigation menu when clicking on mark as read action in the space details panel actions.

### CODE REVIEW

File: component/api/src/main/java/org/exoplatform/social/notification/service/SpaceWebNotificationService.java [hunk 1]
```diff
@@ -61,7 +61,17 @@ public interface SpaceWebNotificationService {
   void markAsRead(SpaceWebNotificationItem notificationItem) throws Exception;
 
   /**
-   * Get a list of aunread items per application to a given {@link MetadataItem} creatorId
+   * Marka a list of unread items per application as read to a given {@link MetadataItem} creatorId
```
Comment (AzmiTouil) [inline component/api/src/main/java/org/exoplatform/social/notification/service/SpaceWebNotificationService.java]:
```suggestion
   * Mark a list of unread items per application as read to a given {@link MetadataItem} creatorId
```

File: webapp/portlet/src/main/webapp/vue-apps/spaces-navigation/components/SpacePanelHamburgerNavigation.vue [hunk 2]
```diff
@@ -82,16 +82,18 @@
         <v-tooltip bottom>
           <template #activator="{ on, attrs }">
             <v-btn
+              :disabled="!spaceUnreadItems || !SpaceWebNotificationsEnabled || !displayBadge"
```
Comment (AzmiTouil) [inline webapp/portlet/src/main/webapp/vue-apps/spaces-navigation/components/SpacePanelHamburgerNavigation.vue]: To add as computed please!

File: webapp/portlet/src/main/webapp/vue-apps/spaces-navigation/components/SpacePanelHamburgerNavigation.vue [hunk 3]
```diff
@@ -127,7 +129,19 @@
               :class="`${applicationIcon(navigation.icon)} icon-default-color icon-default-size`"> </i>
           </v-list-item-icon>
           <v-list-item-content class="mt-n1">
-            {{ navigation.label }}
+            <div class="d-flex align-center justify-space-between">
+              <span>{{ navigation.label }}</span>
+              <v-btn
+                v-if="getUnreadItemsByApplication(navigation.icon) && SpaceWebNotificationsEnabled && displayBadge"
+                class="error-color-background white--text"
+                :class="badgeAnimation && 'zoomOut' || ''"
```
Comment (AzmiTouil) [inline webapp/portlet/src/main/webapp/vue-apps/spaces-navigation/components/SpacePanelHamburgerNavigation.vue:137]: We cannot use some vuetify transitions?

File: component/service/src/main/java/org/exoplatform/social/rest/impl/unread/UnreadRest.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+  @RolesAllowed("users")
+  @Operation(summary = "Delete a metadata items identified by space id", description = "Delete a metadata items identified by space id", method = "DELETE")
+  @ApiResponses(value = {
+      @ApiResponse(responseCode = "204", description = "Request fulfilled"),
+      @ApiResponse(responseCode = "500", description = "Internal server error"),
+      @ApiResponse(responseCode = "400", description = "Invalid query input"),
+      @ApiResponse(responseCode = "404", description = "Not found"),
+      @ApiResponse(responseCode = "401", description = "Unauthorized"),
+  })
+  public Response deleteAllUnreadItems(@Parameter(description = "The space Id", required = true) @PathParam("spaceId") String spaceId, @QueryParam("ignoreNotExisting") boolean ignoreNotExisting) {
```
Comment (AzmiTouil) [inline component/service/src/main/java/org/exoplatform/social/rest/impl/unread/UnreadRest.java]: reformat code please!

**Review by AzmiTouil — CHANGES_REQUESTED** [review]

File: component/service/src/main/java/org/exoplatform/social/rest/impl/unread/UnreadRest.java (hunk 4 above)
Comment (SaraBoutej) [reply component/service/src/main/java/org/exoplatform/social/rest/impl/unread/UnreadRest.java]: Done!

File: webapp/portlet/src/main/webapp/vue-apps/spaces-navigation/components/SpacePanelHamburgerNavigation.vue (hunk 3 above)
Comment (SaraBoutej) [reply webapp/portlet/src/main/webapp/vue-apps/spaces-navigation/components/SpacePanelHamburgerNavigation.vue:137]: I didn't find the zoomOut animation in vuetify transition animation list.

File: webapp/portlet/src/main/webapp/vue-apps/spaces-navigation/components/SpacePanelHamburgerNavigation.vue (hunk 2 above)
Comment (SaraBoutej) [reply webapp/portlet/src/main/webapp/vue-apps/spaces-navigation/components/SpacePanelHamburgerNavigation.vue]: Done

File: component/api/src/main/java/org/exoplatform/social/notification/service/SpaceWebNotificationService.java (hunk 1 above)
Comment (SaraBoutej) [reply component/api/src/main/java/org/exoplatform/social/notification/service/SpaceWebNotificationService.java]: Done!

**Review by AzmiTouil — APPROVED** [review]

**Pull request merged**